### PR TITLE
Floating point hexadecimals

### PR DESCRIPTION
**Pull request by maurobringolf**
This should add support for the number literal format request in #32 . Installed https://github.com/maurobringolf/webassembly-floating-point-hex-parser, extensive testing happens in that repo. This PR just makes a couple of changes to the tokenizer. I think it's not correct in all cases yet but we can iterate on that.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,5 +46,8 @@
     "wasmdump": "./lib/bin/wasmdump.js",
     "wasmast": "./lib/bin/wasmast.js",
     "wasmrun": "./lib/bin/wasmrun.js"
+  },
+  "dependencies": {
+    "webassembly-floating-point-hex-parser": "^0.1.2"
```
Comment (xtuc) [inline package.json]: FYI https://github.com/xtuc/js-webassembly-interpreter/commit/575e72db61b36678ab6dde0433518d54b64ffc1d I had to remove the "no dep" in the README and could we fix the version?

**Comment by xtuc** [comment]
You're actually violating your own licence :smile:, could you include the content of your licence in the file https://github.com/xtuc/js-webassembly-interpreter/blob/master/docs/thirdparty-licenses.txt? Thanks!

File: src/compiler/parsing/watf/tokenizer.js [hunk 2]
```diff
@@ -133,7 +133,7 @@ function tokenize(input: string) {
     }
 
     const NUMBERS = /[0-9|.|_]/;
-    const HEX_NUMBERS = /[0-9|A-F|a-f|_|.]/;
+    const HEX_NUMBERS = /[0-9|A-F|a-f|_|.|p|P|-]/;
```
Comment (xtuc) [inline src/compiler/parsing/watf/tokenizer.js]: How do you parse `nan:0x40000`? Could we add a test for this case?

Comment (maurobringolf) [reply src/compiler/parsing/watf/tokenizer.js]: These are not included in here, I am working on it in a separate branch:

https://github.com/xtuc/js-webassembly-interpreter/tree/nan-payloads

I planned to fix all merge conflicts before PRing from the nan-payloads branch and do this separately.

Comment (xtuc) [reply src/compiler/parsing/watf/tokenizer.js]: ok

**Comment by xtuc** [comment]
Thanks for the changes, we can merge this. Btw i'll upgrade your tests once https://github.com/xtuc/js-webassembly-interpreter/pull/52 is merged

**Pull request merged**